api/src/main/java/org/openmrs/module/ModuleFactory.java

```diff
@@ -574,7 +565,7 @@ public static Module getModuleByPackage(String modulePackage) {
 	 * @see Daemon#startModule(Module)
 	 */
 	public static Module startModule(Module module) throws ModuleException {
-		return Daemon.startModule(module);
```

Comment:
Correct. Have you looked at ModuleFactory.startModules() in reference to your changes? See where is the comment: // Skip module if required ones are not started